src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -638,170 +440,99 @@ public synchronized void open() throws SFException, SnowflakeSQLException {
-        missingProperties.add(addNewDriverProperty(property.getPropertyKey(), null));
-      }
-    }
-    if (isSnowflakeAuthenticator() || isOKTAAuthenticator()) {
-      // userName and password are expected for both Snowflake and Okta.
-      String userName = (String) connectionPropertiesMap.get(SFSessionProperty.USER);
-      if (Strings.isNullOrEmpty(userName)) {
-        missingProperties.add(
-            addNewDriverProperty(SFSessionProperty.USER.getPropertyKey(), "username for account"));
-      }
-
-      String password = (String) connectionPropertiesMap.get(SFSessionProperty.PASSWORD);
-      if (Strings.isNullOrEmpty(password)) {
-        missingProperties.add(
-            addNewDriverProperty(
-                SFSessionProperty.PASSWORD.getPropertyKey(), "password for " + "account"));
-      }
-    }
-
-    boolean useProxy =
-        (boolean) connectionPropertiesMap.getOrDefault(SFSessionProperty.USE_PROXY, false);
-    if (useProxy) {
-      if (!connectionPropertiesMap.containsKey(SFSessionProperty.PROXY_HOST)) {
-        missingProperties.add(
-            addNewDriverProperty(SFSessionProperty.PROXY_HOST.getPropertyKey(), "proxy host name"));
-      }
-      if (!connectionPropertiesMap.containsKey(SFSessionProperty.PROXY_PORT)) {
-        missingProperties.add(
-            addNewDriverProperty(
-                SFSessionProperty.PROXY_PORT.getPropertyKey(),
-                "proxy port; " + "should be an integer"));
-      }
-    }
-    return missingProperties;
-  }
-
-  public String getDatabaseVersion() {
-    return databaseVersion;
+  private boolean isSnowflakeAuthenticator() {
+    Map<SFSessionProperty, Object> connectionPropertiesMap =
+        sessionProperties.getConnectionPropertiesMap();
+    String authenticator = (String) connectionPropertiesMap.get(SFSessionProperty.AUTHENTICATOR);
+    PrivateKey privateKey = (PrivateKey) connectionPropertiesMap.get(SFSessionProperty.PRIVATE_KEY);
+    return (authenticator == null && privateKey == null && privateKeyFileLocation == null)
+        || ClientAuthnDTO.AuthenticatorType.SNOWFLAKE.name().equalsIgnoreCase(authenticator);
   }
 
-  public int getDatabaseMajorVersion() {
-    return databaseMajorVersion;
+  /**
+   * Returns true If authenticator is EXTERNALBROWSER.
+   *
+   * @return true if authenticator type is EXTERNALBROWSER
+   */
+  boolean isExternalbrowserAuthenticator() {
+    Map<SFSessionProperty, Object> connectionPropertiesMap =
+        sessionProperties.getConnectionPropertiesMap();
+    String authenticator = (String) connectionPropertiesMap.get(SFSessionProperty.AUTHENTICATOR);
+    return ClientAuthnDTO.AuthenticatorType.EXTERNALBROWSER.name().equalsIgnoreCase(authenticator);
   }
 
-  public int getDatabaseMinorVersion() {
-    return databaseMinorVersion;
+  /**
+   * Returns true if authenticator is OKTA native
+   *
+   * @return true or false
+   */
+  boolean isOKTAAuthenticator() {
```

Comment:
Why did you introduce these functions

Reply:
I did not. It may appear that way due to the way the diff was processed.